src/System.Drawing.Common/src/System/Drawing/Icon.Unix.cs

```diff
@@ -44,7 +44,9 @@
 
 namespace System.Drawing
 {
-#if !NETCORE
+#if NETCORE
```

Comment:
NETCORE and MONOTOUCH are for the mono repo. We share sources with them. So I would do the following:
```cs
#if netcoreapp
    [System.ComponentModel.TypeConverter("System.Drawing.IconConverter, System.Windows.Extensions, Version=4.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51")]
#endif
#if !NETCORE
#if !MONOTOUCH
    [Editor("System.Drawing.Design.IconEditor, " + Consts.AssemblySystem_Drawing_Design, typeof(System.Drawing.Design.UITypeEditor))]
#endif
    [TypeConverter(typeof(IconConverter))]
#endif
```

or instead of using endif right after  the new attribute you could wrap all of the others with an else.